### ISSUE TITLE
listContents on Windows does not normalize directory separator

### ISSUE THREAD

**Issue by fisharebest** [thephpleague/flysystem#1311]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.0.6

#### Summary

With version 1, the `listContents()` function normalised the directory separator to `/`.

With version 2, the separator depends on the underlying operating system.

#### How to reproduce

```
<?php
// Flysystem 2
include 'vendor/autoload.php';

mkdir('test/foo', 0777, true);
touch('test/foo/bar');

$adapter = new League\Flysystem\Local\LocalFilesystemAdapter('test');
$filesystem = new League\Flysystem\Filesystem($adapter);
$contents = $filesystem->listContents('/', true)->map(fn ($attributes) => $attributes->path())->toArray();
var_dump($contents);
```

On a Linux machine, I get

```
array(2) {
  [0]=>
  string(3) "foo"
  [1]=>
  string(7) "foo/bar"
}
```

On a Windows machine, I get

```
array(2) {
  [0]=>
  string(3) "foo"
  [1]=>
  string(7) "foo\bar"
}
```

In 1.x, both environments would use `/`, as demonstrated by the following script

```
<?php
// Flysystem 1
include 'vendor/autoload.php';

mkdir('test/foo', 0777, true);
touch('test/foo/bar');

$adapter = new League\Flysystem\Adapter\Local('test');
$filesystem = new League\Flysystem\Filesystem($adapter);
$contents = $filesystem->listContents('/', true);
$contents = array_map(fn ($x) => $x['path'], $contents);
var_dump($contents);
```

**Comment by frankdejonge** [comment]
Apart from this being different than in the previous version. Is there anything wrong with this response? Is it causing an error?

**Comment by fisharebest** [comment]
> Is it causing an error?

It causes an error in my application, because I am expecting all filesystems to give consistent results.  

Isn't that the point of a filesystem abstraction library?

Otherwise I need to write an abstraction layer on top of flysystem to hide these differences...

**Comment by frankdejonge** [comment]
What is the error it's causing? And please keep the smart remarks to yourself.

**Comment by fisharebest** [comment]
I have a database that contains references to files.

I have a filesystem which should contain these files.

I need to check for differences - i.e. "missing" and "extra" files.

So, I'm doing a diff on `SELECT filename FROM table` and `$filesystem->listContents()`.

I upgraded from flysystem 1 to flysystem 2, and now my "missing files" page shows every file missing on windows systems.

**Comment by frankdejonge** [comment]
Can you try `dev-2.x`, as in `composer require league/flsysystem:2.x-dev` and see if that fixes it for you?

**Comment by frankdejonge** [comment]
Fixed in 2.0.7

**Comment by fisharebest** [comment]
Thanks for the quick fix.

No intention to be "smart", and apologies if this is how it was received.

Just a literal answer to the question: "Is there anything wrong with this response?"
